Middleware/workflows/managers/workflow_manager.py

```diff
@@ -108,6 +108,16 @@ def __init__(self, workflow_config_name, **kwargs):
         if 'lookbackStartTurn' in kwargs:
             self.lookbackStartTurn = kwargs['lookbackStartTurn']
 
+    def _maybe_update_message_state(self, step_result: Any, current_messages: List[Dict[str, Any]], step_index: int) -> List[Dict[str, Any]]:
```

Comment:
This allows ['messages'] list modification by previous Nodes e.g. when system prompt can be overridden by PythonModule to add additional MCP tools